public/app/features/alerting/unified/Silences.test.tsx

```diff
@@ -270,7 +239,9 @@ describe('Silence edit', () => {
     'creates a new silence',
     async () => {
       renderSilences(baseUrlPath);
-      await waitFor(() => expect(ui.editor.durationField.query()).not.toBeNull());
+      expect(await ui.editor.durationField.find()).toBeInTheDocument();
+
+      const postRequest = waitForServerRequest(silenceCreateHandler());
```

Comment:
Should we not assert on `expect(saveButton).toBeDisabled()` and then `await waitFor(() => expect(saveButton).not.toBeDisabled())`? This is what I've settled on with the settings editor, but had to add a [`delay`](https://mswjs.io/docs/api/delay/) to the POST handler so the button switches disabled state properly.

Reply:
Here I've stuck with asserting that we were returned to the list view, which is what happens for a user on a successful silence creation. WDYT?

Reply:
Oh that's also perfect, do we still need to call `waitForServerRequest` in that case?

Reply:
Would a simple array storing silences help in this case? Each `POST` could add a new item to the array.
Not sophisticated but should work

Reply:
We could do something along those lines, but it would have to be reset between each test using a jest hook, and I'm not sure how best to set it up in a way that scales right now, but also integrates cleanly with the current direction of our mocking approach.

An alternative I thought about was to make the `post` handler do a `server.use(..)` with an overridden list handler, but as we keep our handlers separated from the server (intentionally) it isn't super straightforward to implement right now.

The recommendation [here](https://github.com/mswjs/msw/discussions/964) is to use something like [mswjs/data](https://github.com/mswjs/data), or maintain state yourself (as you suggested).

Do you mind if I defer that decision to a separate PR/later work so we can review that implementation in more detail?

Reply:
Right that sounds like a step too far for now – `waitForServerRequest` certainly has its use here so let's keep it for now :)